server/manager/src/main/java/org/apache/accumulo/manager/metrics/BalancerMetrics.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.manager.metrics;
+
+import java.util.concurrent.atomic.AtomicLong;
+
+import org.apache.accumulo.core.metrics.MetricsProducer;
+
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.MeterRegistry;
+
+public class BalancerMetrics implements MetricsProducer {
+
+  AtomicLong migratingCount = new AtomicLong();
```

Comment:
Could  replace this with a LongSupplier and when creating BalancerMetric pass a lambda in that does `()->migrations.size()`. The `migrations` instance variable in Manager is a synchronized map, so would be safe to call size on it.  With this approach would not need to keep the atomic long in sync with the size of the map.

```suggestion
  LongSupplier migratingCount;
```

Reply:
That's a much better idea than trying to maintain the AtomicLong state. 

I converted it and found that when the metric would first register it would throw a NaN value until balance was first called. 
However, I was able to handle that by checking if the supplier was still null and returning 0 if that was the case.